### PR TITLE
#910: Consolidate LengthOf classes

### PR DESCRIPTION
**Pull request by dgroup**
This PR for #910 contains the following changes: 
 - New scalar `LengthOf` 
 - Deleted `o.c.iterable.LengthOf`, `o.c.iterator.LengthOf` and `o.c.io.LengthOf`, 
 - The LengthOfTest with joined unit tests for `o.c.iterable.LengthOf`, `o.c.iterator.LengthOf` and `o.c.io.LengthOf` 
 - Changed `import`(s) section in order to use the `org.cactoos.scalar.LengthOf`

### CODE REVIEW

**Comment by 0crat** [comment]
Job [#987](https://github.com/yegor256/cactoos/issues/987) is now in scope, role is `REV`

**Comment by dgroup** [comment]
@llorllale, please review the PR.

**Comment by llorllale** [comment]
@dgroup 35 files!

File: src/main/java/org/cactoos/scalar/LengthOf.java [hunk 1]
```diff
@@ -21,14 +21,16 @@
...
 
 import java.io.InputStream;
+import java.util.Iterator;
 import org.cactoos.Input;
-import org.cactoos.scalar.NumberEnvelope;
+import org.cactoos.Scalar;
 
 /**
- * Length of {@link Input}.
+ * Length of {@link Iterator}, {@link Iterable} or {@link Input}.
```
Comment (llorllale) [inline src/main/java/org/cactoos/scalar/LengthOf.java]: @dgroup just "Length" suffices I think.

File: src/main/java/org/cactoos/scalar/LengthOf.java [hunk 2]
```diff
@@ -62,7 +90,7 @@ public LengthOf(final Input input) {
         }
     )
     public LengthOf(final Input input, final int max) {
-        super(() -> {
+        this(() -> {
             if (max == 0) {
                 throw new IllegalArgumentException(
                     "Cannot use a buffer limited to zero size"
```
Comment (llorllale) [inline src/main/java/org/cactoos/scalar/LengthOf.java]: @dgroup I think this msg should be changed now

**Comment by llorllale** [comment]
@dgroup left you a couple of comments, please check.

File: src/main/java/org/cactoos/scalar/LengthOf.java (hunk 1 above)
Comment (dgroup) [reply src/main/java/org/cactoos/scalar/LengthOf.java]: @llorllale, agree.

File: src/main/java/org/cactoos/scalar/LengthOf.java (hunk 2 above)
Comment (dgroup) [reply src/main/java/org/cactoos/scalar/LengthOf.java]: @llorllale, sorry, didn't get what you mean.
The `max` field is buffer size, therefore if user pass the `0` he should receive the exception related to `zero-buffer-size`

Comment (llorllale) [reply src/main/java/org/cactoos/scalar/LengthOf.java]: @dgroup sorry, my bad - I was not paying attention to the ctor this msg is in. Please disregard my comment.

**Comment by dgroup** [comment]
> @dgroup 35 files!
@llorllale yes, 35 files. I saw your comment about 10 files per pull request after raising of current.
I'll follow it from the next PR.
If you'll check most of these classes have only 1 change related to `import` of `LengthOf` so from my perspective isn't so complex to review/understand. Maybe I'm wrong.

File: src/main/java/org/cactoos/scalar/LengthOf.java (hunk 2 above)
Comment (dgroup) [reply src/main/java/org/cactoos/scalar/LengthOf.java]: @llorllale, thank you, I think you missed it due to huge PR.
Sorry about that.

**Comment by llorllale** [comment]
@rultor merge

**Comment by 0crat** [comment]
Job `gh:yegor256/cactoos#987` is not assigned, can't get performer

**Comment by 0crat** [comment]
The job [#987](https://github.com/yegor256/cactoos/issues/987) is now out of scope

**Pull request merged**